core-lib/src/main/java/org/idpass/smartscanner/lib/SmartScannerActivity.kt

```diff
@@ -770,27 +843,108 @@ class SmartScannerActivity : BaseActivity(), OnClickListener {
-                    ).roundToInt()
-                    rectangleGuide?.layoutParams?.width = nWidth
-                }
+            showMRZGuide()
+        } else  if (config != null && config?.showOcrGuide == true) {
+            showOCRGuide()
+        } else {
+            ocrGuideContainer?.alpha = 0f
+        }
+    }
+
+    private fun showMRZGuide() {
+        ocrGuideContainer?.alpha = 1f
 
-                // if height guide is not by default
-                if (conf.heightGuide != 70) {
-                    val nHeight = TypedValue.applyDimension(
-                        TypedValue.COMPLEX_UNIT_DIP, conf.heightGuide.toFloat(), resources.displayMetrics
+        config?.let { conf ->
+            viewFinder.post {
+                val width = if (conf.widthGuide == 0) {
+                    //set the default width
+                   (viewFinder.width - 21.toPx)
+                } else {
+                    TypedValue.applyDimension(
+                        TypedValue.COMPLEX_UNIT_DIP,
+                        conf.widthGuide.toFloat(),
+                        resources.displayMetrics
                     ).roundToInt()
-                    rectangleGuide?.layoutParams?.height = nHeight
                 }
-                rectangleGuide?.requestLayout()
+
+                rectangleGuide?.layoutParams?.width = width
+                widthGuideOCR?.layoutParams?.width = width
+
+                //set height
+                val nHeight = TypedValue.applyDimension(
+                    TypedValue.COMPLEX_UNIT_DIP,
+                    conf.heightGuide.toFloat(),
+                    resources.displayMetrics
+                ).roundToInt()
+                rectangleGuide?.layoutParams?.height = nHeight
+
+                //set default position
+
+                val xPercentage = (viewFinder.width - width).div(2)
+                val yPercentage = viewFinder.height - nHeight - 30.toPx
+
+                xGuideView?.layoutParams?.width = xPercentage
+                yGuideView?.layoutParams?.height = yPercentage
+
+                xGuideView?.requestLayout()
+                yGuideView?.requestLayout()
+            }
+            rectangleGuide?.requestLayout()
+        }
+    }
+    private fun showOCRGuide() {
+        ocrGuideContainer?.alpha = 1f
+
+        config?.let { conf ->
+            if (conf.widthGuide != 0) {
+                val nWidth = TypedValue.applyDimension(
+                    TypedValue.COMPLEX_UNIT_DIP,
+                    conf.widthGuide.toFloat(),
+                    resources.displayMetrics
+                ).roundToInt()
+                rectangleGuide?.layoutParams?.width = nWidth
+                widthGuideOCR?.layoutParams?.width = nWidth
             }
 
-        } else {
-            rectangleGuide?.alpha = 0f;
+            // if height guide is not by default
+            if (conf.heightGuide != 70) {
```

Comment:
if compare to default value, it would be better to put that into a constant and use it

Reply:
Fixed